main.cpp

```diff
@@ -854,14 +854,22 @@ struct file_memory_access : public memory_access {
 
     void read(uint32_t address, uint8_t *buffer, uint32_t size) override {
         while (size) {
-            auto result = rmap.get(address);
-            uint this_size = std::min(size, result.first.max_offset - result.first.offset);
-            assert( this_size);
-            fseek(file, result.second + result.first.offset, SEEK_SET);
-            fread(buffer, this_size, 1, file);
-            buffer += this_size;
-            address += this_size;
-            size -= this_size;
+            try {
+                auto result = rmap.get(address);
+                uint this_size = std::min(size, result.first.max_offset - result.first.offset);
+                printf("this_size is 0x%08x\n", this_size);
```

Comment:
Is this a leftover debugging-print?

Reply:
Ok, removed the printf statement and made the gap-spanning coalescing conditional on the memory type.